payments-core/src/main/java/com/stripe/android/payments/core/authentication/threeds2/Stripe3ds2TransactionViewModel.kt

```diff
@@ -35,20 +32,48 @@ import com.stripe.android.stripe3ds2.transaction.Transaction
 import kotlinx.coroutines.Dispatchers
 import kotlinx.coroutines.delay
 import kotlinx.coroutines.withContext
+import javax.inject.Inject
 import kotlin.coroutines.CoroutineContext
 
 internal class Stripe3ds2TransactionViewModel(
     private val args: Stripe3ds2TransactionContract.Args,
-    private val stripeRepository: StripeRepository,
-    private val analyticsRequestExecutor: AnalyticsRequestExecutor,
-    private val analyticsRequestFactory: AnalyticsRequestFactory,
-    private val threeDs2Service: StripeThreeDs2Service,
-    private val messageVersionRegistry: MessageVersionRegistry,
-    private val challengeResultProcessor: Stripe3ds2ChallengeResultProcessor,
     private val initChallengeRepository: InitChallengeRepository,
-    private val workContext: CoroutineContext,
     private val savedStateHandle: SavedStateHandle
 ) : ViewModel() {
+
+    @Inject
+    lateinit var stripeRepository: StripeRepository
+
+    @Inject
+    lateinit var analyticsRequestExecutor: AnalyticsRequestExecutor
+
+    @Inject
+    lateinit var analyticsRequestFactory: AnalyticsRequestFactory
+
+    @Inject
+    lateinit var messageVersionRegistry: MessageVersionRegistry
+
+    @Inject
+    lateinit var threeDs2Service: StripeThreeDs2Service
+
+    @Inject
+    lateinit var challengeResultProcessor: Stripe3ds2ChallengeResultProcessor
+
+    @Inject
+    @IOContext
+    lateinit var workContext: CoroutineContext
```

Comment:
This feels like we're going in the wrong direction. Why do we need to move from constructor injection to field injection?

Reply:
Constructor injection only works if this object is created along with Component, like `Stripe3DS2Authenticator`. 
Field injection is used for objects created _after_ the Component, in which case a `Component#inject(FieldInjectedObject)` call is made to instantiated the field object(and hence they are defined lateinit). 

In this case, a new `Stripe3ds2TransactionViewModel` is created each time `Stripe3DS2Authenticator#authenticate` is called, therefore we're using the component hosting `Stripe3DS2Authenticator` to inject into this newly created ViewModel each time

Reply:
I don't get "Field injection is used for objects created after the Component". The component should be able to instantiate an object at any time, right? Can't you inject a `Provider` and instantiate a new `Stripe3ds2TransactionViewModel` every time you need?